### DISCUSSION TITLE
Working Pipeline Code - Using an Azure DevOps Pipeline to deploy WebApp

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/sample-app-aoai-chatGPT#665] [Show and tell]
`trigger:
  branches:
    include:
    - develop

variables:
  azureServiceConnectionId: "<AzureSubscription>"
  webAppName: "<AppServiceName>"
  vmImageName: "ubuntu-latest"
  environmentName: "<Environment>"
  projectRoot: $(System.DefaultWorkingDirectory)

stages:
- stage: Build
  displayName: Build stage
  jobs:
  - job: BuildJob
    pool:
      vmImage: $(vmImageName)
    steps:
    - task: UsePythonVersion@0
      displayName: 'Use Python'

    - script: |
        python -m venv antenv
        source antenv/bin/activate
        python -m pip install --upgrade pip
        pip install setup
        pip install -r requirements.txt
      workingDirectory: $(projectRoot)
      displayName: "Install Python Requirements"

    - task: NodeTool@0
      inputs:
        versionSpec: '20.x'
      displayName: 'Install Node.js'

    - script: |
        cd ./frontend
        npm install
        npm run build
        cd ..
        rm -rf $(projectRoot)/frontend/node_modules
      displayName: 'npm install and build, removing node_modules'

    - task: ArchiveFiles@2
      displayName: 'Archive files'
      inputs:
        rootFolderOrFile: '$(projectRoot)'
        includeRootFolder: false
        archiveType: zip
        archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
        replaceExistingArchive: true

    - upload: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
      displayName: 'Upload package'
      artifact: drop

- stage: Deploy
  displayName: 'Deploy Web App'
  dependsOn: Build
  condition: succeeded()
  jobs:
  - deployment: DeploymentJob
    pool:
      vmImage: $(vmImageName)
    environment: $(environmentName)
    strategy:
      runOnce:
        deploy:
          steps:
          - task: UsePythonVersion@0
            displayName: 'Use Python version'
          - task: AzureWebApp@1
            displayName: 'Deploy Azure Web App : {{ webAppName }}'
            inputs:
              azureSubscription: $(azureServiceConnectionId)
              appName: $(webAppName)
              package: $(Pipeline.Workspace)/drop/$(Build.BuildId).zip`

**Comment by tgk35** [comment]
do i have to push the .env file also to the azure repo to use this pipeline file?

**Comment by tgk35** [comment]
can you give the sample values of the variables of variables you have used?
